tools/querytee/response_comparator.go

```diff
@@ -173,26 +179,30 @@ func compareVector(expectedRaw, actualRaw json.RawMessage, opts SampleComparison
 			Value:     actualMetric.Value,
 		}, opts)
 		if err != nil {
-			return errors.Wrapf(err, "sample pair not matching for metric %s", expectedMetric.Metric)
+			return nil, errors.Wrapf(err, "sample pair not matching for metric %s", expectedMetric.Metric)
 		}
 	}
 
-	return nil
+	if len(missingMetrics) > 0 {
```

Comment:
This is only showing the first one, but there may be more - right? Might be good to include the length

Reply:
Yes. The original behaviour is to only show the first one. I'm happy to use a multi error.

Reply:
Done :slightly_smiling_face: